src/main/java/net/imglib2/roi/labeling/OverlappingLabels.java

```diff
@@ -0,0 +1,194 @@
+{
+
+	private Img< UnsignedIntType > overlapMatrix;
+
+	private List< T > labelList;
+
+	private Img< LongType > labelSizeImg;
+
+	private Img< DoubleType > relativeOverlapMatrix;
+
+	public OverlappingLabels( RandomAccessibleInterval< LabelingType< T > > labeling )
+	{
+
+		FragmentLabelRegions labelRegions = new FragmentLabelRegions( labeling );
+
+		// List of existing labels
+		// NB: this triggers labelRegions computation
+		labelList = new ArrayList<>( labelRegions.getExistingLabels() );
+		Collections.sort( labelList );
+
+		// Create 2D matrix image
+		FinalInterval matrixDims = new FinalInterval( labelList.size(), labelList.size() );
+		overlapMatrix = Util.getSuitableImgFactory( matrixDims, new UnsignedIntType() ).create( matrixDims );
+
+		// Sum up sizes of single fragments
+		RandomAccess< UnsignedIntType > randomAccess = overlapMatrix.randomAccess();
+		UnsignedIntType sizeType = new UnsignedIntType();
+		LabelingMapping< T > labelingMapping = Labelings.getLabelingMapping( labeling );
+		for ( int i = 0; i < labelingMapping.numSets(); i++ )
+		{
+			Set< T > currentLabels = labelingMapping.labelsAtIndex( i );
+			long size = labelRegions.getFragmentSizeForIndex( i );
+			if ( size > 0 )
+			{
+				sizeType.set( size );
+				for ( T a : currentLabels )
+				{
+					randomAccess.setPosition( labelList.indexOf( a ), 0 );
+					for ( T b : currentLabels )
+					{
+						randomAccess.setPosition( labelList.indexOf( b ), 1 );
+						randomAccess.get().add( sizeType );
+					}
+				}
+			}
+		}
+
+		labelSizeImg = ArrayImgs.longs( labelList.size() );
+		Cursor< LongType > cursor = labelSizeImg.localizingCursor();
+		while ( cursor.hasNext() )
+		{
+			cursor.next().set( labelRegions.getLabelRegion( labelList.get( cursor.getIntPosition( 0 ) ) ).size() );
+		}
+
+		relativeOverlapMatrix = Util.getSuitableImgFactory( matrixDims, new DoubleType() ).create( matrixDims );
+		IntervalView< LongType > divisor = Views.addDimension( labelSizeImg, 0, labelList.size() - 1 );
+
+		LoopBuilder.setImages( overlapMatrix, divisor, relativeOverlapMatrix ).forEachPixel( ( a, b, c ) -> c.set( ( double ) a.get() / b.get() ) );
+	}
+
+	/**
+	 * Get the confusion/co-occurrence matrix.
+	 * 
+	 * Values indicate the number of pixels that are in common between the label
+	 * at position x and the label at position y. For the mapping of dimension
+	 * indices to labels, see {@link OverlappingLabels#getIndexedLabels()}.
+	 * 
+	 * @return the co-occurrence matrix
+	 */
+	public Img< UnsignedIntType > getMatrixImg()
+	{
+		return overlapMatrix;
+	}
+
+	/**
+	 * Get the normalized directional confusion matrix.
+	 * 
+	 * Values indicate the proportion of the label at position x that is
+	 * co-occurring with the label at position y. For the mapping of dimension
+	 * indices to labels, see {@link OverlappingLabels#getIndexedLabels()}.
+	 * 
+	 * @return the normalized co-occurrence matrix
+	 */
+	public Img< DoubleType > getNormalizedMatrixImg()
+	{
```

Comment:
I would prefer to return RandomAccessibleInterval here. Is there any reason not to?

Reply:
Changed and renamed to `getNormalizedMatrix`.